firefox_puppeteer/api/places.py

```diff
@@ -88,14 +88,22 @@ def restore_default_bookmarks(self):
 
               let observer = {
                 observe: function (aSubject, aTopic, aData) {
-                  Services.obs.removeObserver(observer, "bookmarks-restore-success");
+                  if ( aTopic === "bookmarks-restore-success") {
+                    Services.obs.removeObserver(observer, "bookmarks-restore-success");
 
-                  marionetteScriptFinished(true);
+                    marionetteScriptFinished(true);
+                  }
+                  if ( aTopic == "bookmarks-restore-failed") {
+                    Services.obs.removeObserver(observer, "bookmars-restore-failed");
+
+                    marionetteScriptFinished(false);
```

Comment:
You can get rid of the above two if blocks if you do a simple compare of the topic with `bookmarks-restore-success` and take this as argument for marionetteScriptFinished.